### PR TITLE
[MANA-56] memory access tracker util

### PR DESCRIPTION
**Pull request by jungan**
Add a tool to track memory access at a more gradual level.

Tested with a sample program and works well.

### CODE REVIEW

**Comment by gc00** [comment]
I think this will be difficult for us to review carefully.  I suggest that you check this in, and if you find bugs later (or enhancements), you can do a second PR.

Second, I suggest a copyright statement to formally declare this open source with you as the author.  You only nejed this for the .c and .h files.  For a model, see:
`util/gdb-dmtcp-utils`

I am approving this now, but please add the copyright statement before pushing it in.

File: util/memory_access_tracker/memory_access_tracker.c [hunk 1]
```diff
@@ -0,0 +1,239 @@
+#include <stdio.h>
```
Comment (gc00) [inline util/memory_access_tracker/memory_access_tracker.c:1]: Above this line, please add a copyright statement to formally declare this open source with you as the author. You only nejed this for the .c and .h files. For a model, see:
`util/gdb-dmtcp-utils`

File: util/memory_access_tracker/test_memory_tracker.c [hunk 2]
```diff
@@ -0,0 +1,62 @@
+#include <unistd.h>
```
Comment (gc00) [inline util/memory_access_tracker/test_memory_tracker.c:1]: Above this line, please add a copyright statement to formally declare this open source with you as the author. You only nejed this for the .c and .h files. For a model, see:
`util/gdb-dmtcp-utils`

**Review by gc00 — APPROVED** [review]
Please add a copyright statement to formally declare this open source with you as the author. You only nejed this for the .c and .h files. For a model, see:
`util/gdb-dmtcp-utils`

I'm approving this in advance of you adding the copyright statement.  The code seems reasonable, and we can always add bug fixes and enhancements later as we discover them.

This would eventually also make a great PR for DMTCP itself.  You should have permission to submit a PR to DMTCP.

I

**Comment by jungan** [comment]
test this please

**Pull request merged**